homeassistant/components/velux/binary_sensor.py

```diff
@@ -0,0 +1,77 @@
+"""Support for rain sensors build into some velux windows."""
+
+from __future__ import annotations
+
+from datetime import timedelta
+
+from pyvlx.exception import PyVLXException
+from pyvlx.opening_device import OpeningDevice, Window
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorDeviceClass,
+    BinarySensorEntity,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+
+from .const import DOMAIN, LOGGER
+from .entity import VeluxEntity
+
+PARALLEL_UPDATES = 1
+SCAN_INTERVAL = timedelta(minutes=5)  # Use standard polling
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config: ConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Set up rain sensor(s) for Velux platform."""
+    module = hass.data[DOMAIN][config.entry_id]
+
+    entities = [
+        VeluxRainSensor(node, config.entry_id)
+        for node in module.pyvlx.nodes
+        if isinstance(node, Window) and node.rain_sensor
+    ]
+
+    async_add_entities(entities)
+
+
+class VeluxRainSensor(VeluxEntity, BinarySensorEntity):
+    """Representation of a Velux rain sensor."""
+
+    node: Window
+    _attr_should_poll = True  # the rain sensor / opening limitations needs polling unlike the rest of the Velux devices
+    _attr_entity_registry_enabled_default = False
+
+    def __init__(self, node: OpeningDevice, config_entry_id: str) -> None:
+        """Initialize VeluxRainSensor."""
+        super().__init__(node, config_entry_id)
+        LOGGER.info("Creating velux rain sensor from %s", node.name)
+        self._attr_unique_id = f"{self._attr_unique_id}_rain_sensor"
+        self._attr_name = f"{node.name} Rain Sensor"
+        self._attr_device_class = BinarySensorDeviceClass.MOISTURE
```

Comment:
This can be set outside of the constructor

Reply:
done